db/db_impl/db_impl_write.cc

```diff
@@ -1124,13 +1124,28 @@ IOStatus DBImpl::WriteToWAL(const WriteThread::WriteGroup& write_group,
     //    writer thread, so no one will push to logs_,
     //  - as long as other threads don't modify it, it's safe to read
     //    from std::deque from multiple threads concurrently.
+    //
+    // Sync operation should work with locked log_write_mutex_, because:
+    //   when DBOptions.manual_wal_flush_ is set,
+    //   Flushwal function will be invoked by another thread.
+    //   if log file sync operation with unlocked log_write_mutex_,
+    //   log file may be got a data corruption or duplicate record.
```

Comment:
Nit: the log file may get data corruption

Reply:
Is 'duplicate record' a confirmed case? If the behavior is undefined, I think we can just say "data corruption".

Reply:
> Is 'duplicate record' a confirmed case? If the behavior is undefined, I think we can just say "data corruption".

yes, the duplicate record is confirmed case. In the db_test.cc, when (splitWriteAndSlowSecondWrite: false, manual_wal_flush: true, sync: true), the test always failed by duplicate record.